File: uploadfile-android/app/src/main/java/com/newgate/uploadfile/model/FileUploadResponse.kt
package com.newgate.uploadfile.model

import com.google.gson.annotations.SerializedName

/**
 * Created by apple on 2/21/18.
 */
data class FileUploadResponse(
        @SerializedName("message") var message: String,
        @SerializedName("data") var data: FileUploadData)

data class FileUploadsResponse(
        @SerializedName("message") var message: String,
        @SerializedName("data") var data: ArrayList<FileUploadData>)

data class FileUploadData(
        @SerializedName("originalname") var originalname: String,
        @SerializedName("uploadname") var uploadname: String)

data class FileUpload(
        var name: String,
        var progress: Int,
        var uploadname: String = "",
        var displayViewDetail: Boolean = false,
        var beginUpdate: Boolean = false)

File: uploadfile-ios/uploadfile-ios/model/FileUploadResponse.swift
//
//  FileUploadData.swift
//  uploadfile-ios
//
//  Created by Nguyen on 2/24/18.
//  Copyright © 2018 Apple. All rights reserved.
//

import UIKit
import SwiftyJSON

class FileUploadResponse: NSObject {
    var message: String?

    var data: FileUploadData?

    init(json: JSON) {
        message = json["message"].string
        data = FileUploadData(json: json["data"])
    }
}

class FileUploadArrayResponse: NSObject {
    var message: String?

    var datas: Array<FileUploadData>?

    init(json: JSON) {
        message = json["message"].string
        if let dataJsons = json["data"].array {
            datas = Array()
            for dataJson in dataJsons {
                let data = FileUploadData(json: dataJson)
                datas?.append(data)
            }
        }
    }
}

class FileUploadData: NSObject {
    var originalname: String?

    var uploadname: String?

    init(json: JSON) {
        originalname = json["originalname"].string
        uploadname = json["uploadname"].string
    }
}
